app.py

```diff
@@ -75,104 +79,156 @@ def voinfo():
         active_value = args.get("active_value", "")
         if active_value == "0":
-            for rg in rgs:
-                rg["Resources"]["Resource"] = [r for r in ensure_list(rg["Resources"]["Resource"]) if not r["Active"]]
+            filters.active = False
         elif active_value == "1":
-            for rg in rgs:
-                rg["Resources"]["Resource"] = [r for r in ensure_list(rg["Resources"]["Resource"]) if r["Active"]]
+            filters.active = True
         else:
-            return Response("Invalid arguments: active_value must be 0 or 1", status=400)
+            raise InvalidArgumentsError("active_value must be 0 or 1")
     if "disable" in args:
         disable_value = args.get("disable_value", "")
         if disable_value == "0":
-            for rg in rgs:
-                rg["Resources"]["Resource"] = [r for r in ensure_list(rg["Resources"]["Resource"]) if not r["Disable"]]
+            filters.disable = False
         elif disable_value == "1":
-            for rg in rgs:
-                rg["Resources"]["Resource"] = [r for r in ensure_list(rg["Resources"]["Resource"]) if r["Disable"]]
+            filters.disable = True
         else:
-            return Response("Invalid arguments: disable_value must be 0 or 1", status=400)
-
+            raise InvalidArgumentsError("disable_value must be 0 or 1")
     if "gridtype" in args:
         gridtype_1, gridtype_2 = args.get("gridtype_1", ""), args.get("gridtype_2", "")
         if gridtype_1 == "on" and gridtype_2 == "on":
             pass
         elif gridtype_1 == "on":
-            rgsummary["ResourceSummary"]["ResourceGroup"] = [rg for rg in rgs if
-                                                             rg["GridType"] == "OSG Production Resource"]
+            filters.grid_type = GRIDTYPE_1
         elif gridtype_2 == "on":
-            rgsummary["ResourceSummary"]["ResourceGroup"] = [rg for rg in rgs if
-                                                             rg["GridType"] == "OSG Integration Test Bed Resource"]
+            filters.grid_type = GRIDTYPE_2
         else:
-            # invalid arguments: no RGs for you!
-            return Response("Invalid arguments: gridtype_1 or gridtype_2 or both must be \"on\"", status=400)
+            raise InvalidArgumentsError("gridtype_1 or gridtype_2 or both must be \"on\"")
+    if "downtime_attrs_showpast" in args:
+        # doesn't make sense for rgsummary but will be ignored anyway
+        try:
+            v = args["downtime_attrs_showpast"]
+            if v == "all":
+                filters.past_days = -1
+            elif not v:
+                filters.past_days = 0
+            else:
+                filters.past_days = int(args["downtime_attrs_showpast"])
+        except ValueError:
+            raise InvalidArgumentsError("downtime_attrs_showpast must be an integer, \"\", or \"all\"")
+
+    # 2 ways to filter by service: either pass service_1=on, service_2=on, etc.
```

Comment:
Should this comment be more generalized? The following code doesn't seem to just apply to services

Reply:
yeah, let me fix that.